cylc/flow/task_pool.py

```diff
@@ -1453,16 +1482,22 @@ def force_trigger_tasks(
                 self._get_main_task_by_id(task_id)
                 or self._get_hidden_task_by_id(task_id)
             )
+            # flow values already checked by the trigger client.
             if itask is None:
-                # Spawn with new flow number, unless no reflow.
-                if reflow:
+                if flow[0] == FLOW_ALL:
+                    flow_nums = self._get_active_flow_nums()
+                elif flow[0] == FLOW_NEW:
                     flow_nums = {self.flow_mgr.get_new_flow(flow_descr)}
-                else:
+                elif flow[0] == FLOW_NONE:
                     flow_nums = set()
-                itask = self.spawn_task(name, point, flow_nums)
+                else:
+                    flow_nums = {int(n) for n in flow}
```

Comment:
I think this gets pre-validated(?) but should still probably still stick this in a try/except for safety as a ValueError here would tear down the workflow.